<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="ie=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="author"
    content="map[name:Pei-Wen Shih]">
<meta name="description"
    content="Today goals  Things you should know before assembly  Fastq format Fasta format GC- content Status of genome  Step to bacterial sequencing analysis  Things you should know before assembly Fastq format start with @
@cc3e68c4-b53d-43da-be7e-b961113007e2 -&amp;gt; sequence name ATCCGGAATCGGTTACTGTTGGGAACCTTTGC -&amp;gt; sequence &#43; -&amp;gt; quality line break #%(*))&#43;&#43;.2/148447;7&#43;001./18-7-,,30&amp;amp;*2 -&amp;gt; quality score  Fasta format start with &amp;gt;
&amp;gt;tig00000001	-&amp;gt; sequence name TGATAAAAGTATTCATATAATCTCCTATCATTTCAAAATTTAAT -&amp;gt; sequence &amp;gt;tig00000002	ATATTAGTGTGTCTATTTTATGGGGCTAGGAAAGGAGGTACATT  GC-content the percentage of nitrogenous bases on a DNA or RNA molecule that are either guanine or cytosine $$ \frac{G&#43;C}{A&#43;T&#43;G&#43;C} \times 100 \% $$" />
<meta name="keywords" content="" />
<meta name="robots" content="noodp" />
<link rel="canonical" href="https://spaert.github.io/posts/insights-into-bacterial-genome-sequence-analysis/" />


<title>
    
    Insights into Bacterial Genome Sequence Analysis :: Hello World 
    
</title>



<link href="https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.2.1/css/flag-icon.min.css" rel="stylesheet"
    type="text/css">



<link rel="stylesheet" href="/main.min.d90905eb29b71569a516566cd9662312bee9ab021c760de6949cdc92d6688be9.css">



<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#252627">
<link rel="shortcut icon" href="/favicon.ico">
<meta name="theme-color" content="#252627">
<meta itemprop="name" content="Insights into Bacterial Genome Sequence Analysis">
<meta itemprop="description" content="Today goals  Things you should know before assembly  Fastq format Fasta format GC- content Status of genome  Step to bacterial sequencing analysis  Things you should know before assembly Fastq format start with @
@cc3e68c4-b53d-43da-be7e-b961113007e2 -&gt; sequence name ATCCGGAATCGGTTACTGTTGGGAACCTTTGC -&gt; sequence &#43; -&gt; quality line break #%(*))&#43;&#43;.2/148447;7&#43;001./18-7-,,30&amp;*2 -&gt; quality score  Fasta format start with &gt;
&gt;tig00000001	-&gt; sequence name TGATAAAAGTATTCATATAATCTCCTATCATTTCAAAATTTAAT -&gt; sequence &gt;tig00000002	ATATTAGTGTGTCTATTTTATGGGGCTAGGAAAGGAGGTACATT  GC-content the percentage of nitrogenous bases on a DNA or RNA molecule that are either guanine or cytosine $$ \frac{G&#43;C}{A&#43;T&#43;G&#43;C} \times 100 \% $$">


<meta itemprop="datePublished" content="2019-06-09T21:51:13&#43;08:00" />
<meta itemprop="dateModified" content="2019-06-09T21:51:13&#43;08:00" />
<meta itemprop="wordCount" content="348">



<meta itemprop="keywords" content="genome assembly,third generation sequence," />
<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="https://spaert.github.io"/>

<meta name="twitter:title" content="Insights into Bacterial Genome Sequence Analysis"/>
<meta name="twitter:description" content="Today goals  Things you should know before assembly  Fastq format Fasta format GC- content Status of genome  Step to bacterial sequencing analysis  Things you should know before assembly Fastq format start with @
@cc3e68c4-b53d-43da-be7e-b961113007e2 -&gt; sequence name ATCCGGAATCGGTTACTGTTGGGAACCTTTGC -&gt; sequence &#43; -&gt; quality line break #%(*))&#43;&#43;.2/148447;7&#43;001./18-7-,,30&amp;*2 -&gt; quality score  Fasta format start with &gt;
&gt;tig00000001	-&gt; sequence name TGATAAAAGTATTCATATAATCTCCTATCATTTCAAAATTTAAT -&gt; sequence &gt;tig00000002	ATATTAGTGTGTCTATTTTATGGGGCTAGGAAAGGAGGTACATT  GC-content the percentage of nitrogenous bases on a DNA or RNA molecule that are either guanine or cytosine $$ \frac{G&#43;C}{A&#43;T&#43;G&#43;C} \times 100 \% $$"/>




<meta property="article:published_time" content="2019-06-09 21:51:13 &#43;0800 CST" />






        
    </head>

    <body class="">
        <div class="container">
            <header class="header">
    <span class="header__inner">
        <a href="/" style="text-decoration: none;">
    <div class="logo">
        
            <span class="logo__mark">></span>
            <span class="logo__text">$cd /home/pei-wen</span>
            <span class="logo__cursor"></span>
        
    </div>
</a>


        <span class="header__right">
            
                <nav class="menu">
    <ul class="menu__inner"><li><a href="https://spaert.github.io/about">About</a></li><li><a href="https://spaert.github.io/posts">Blog</a></li>
    </ul>
</nav>

                <span class="menu-trigger">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
                        <path d="M0 0h24v24H0z" fill="none"/>
                        <path d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"/>
                    </svg>
                </span>
            

            <span class="theme-toggle"><svg class="theme-toggler" width="24" height="24" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M22 41C32.4934 41 41 32.4934 41 22C41 11.5066 32.4934 3 22
  3C11.5066 3 3 11.5066 3 22C3 32.4934 11.5066 41 22 41ZM7 22C7
  13.7157 13.7157 7 22 7V37C13.7157 37 7 30.2843 7 22Z"/>
</svg>
</span>
        </span>
    </span>
</header>


            <div class="content">
                
    <main class="post">

        <div class="post-info">
            <p><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-clock"><circle cx="12" cy="12" r="10"></circle><polyline points="12 6 12 12 16 14"></polyline></svg>2 minutes

            

            </p>
        </div>

        <article>
            <h1 class="post-title"><a href="https://spaert.github.io/posts/insights-into-bacterial-genome-sequence-analysis/">Insights into Bacterial Genome Sequence Analysis</a></h1>
                <hr />
                <aside id="toc">
                <div class="toc-title">Table of Contents</div>
                    <nav id="TableOfContents">
<ul>
<li><a href="#today-goals">Today goals</a></li>
<li><a href="#things-you-should-know-before-assembly">Things you should know before assembly</a>
<ul>
<li><a href="#fastq-format">Fastq format</a></li>
<li><a href="#fasta-format">Fasta format</a></li>
<li><a href="#gc-content">GC-content</a></li>
<li><a href="#satuts-of-genome">Satuts of genome</a>
<ul>
<li><a href="#1-contig">1. contig</a></li>
<li><a href="#2-scaffold">2. scaffold</a></li>
<li><a href="#3-complete">3. complete</a></li>
</ul></li>
</ul></li>
<li><a href="#step-to-bacterial-sequencing-analysis">Step to bacterial sequencing analysis</a>
<ul>
<li><a href="#step-1-quality-control-assessing-the-quality-of-tgs">Step 1. Quality Control - Assessing the quality of TGS</a>
<ul>
<li><a href="#1-1-checking-raw-read-statistics">1.1 Checking raw read statistics</a></li>
<li><a href="#1-2-qc-report">1.2 QC report</a></li>
</ul></li>
<li><a href="#step-2-trimming-filtering-data-preprocessing-of-raw-data-optional">Step 2. Trimming Filtering Data - Preprocessing of raw data (optional)</a>
<ul>
<li><a href="#2-1-filter-raw-read">2.1 Filter raw read</a></li>
<li><a href="#2-2-demultiplexing-trimming-adapter-barcode">2.2 Demultiplexing, Trimming adapter (barcode)</a></li>
</ul></li>
<li><a href="#step-3-sequence-assembly-long-read-genome-assembly">Step 3. Sequence Assembly - Long read genome assembly</a>
<ul>
<li><a href="#3-1-de-novo-assembly">3.1 De novo assembly</a></li>
</ul></li>
<li><a href="#step-4-assembly-validation">Step 4. Assembly Validation</a>
<ul>
<li><a href="#4-1-assembly-evaluation">4.1 Assembly evaluation</a></li>
<li><a href="#4-2-assembly-status-graph">4.2 Assembly status graph</a></li>
<li><a href="#4-3-sequence-similarity-search">4.3 Sequence Similarity Search</a></li>
<li><a href="#4-4-quality-of-genomes">4.4 Quality of genomes</a></li>
</ul></li>
</ul></li>
</ul>
</nav>
                </aside>
                <hr />

            

            <div class="post-content">
                

<h1 id="today-goals">Today goals</h1>

<ul>
<li>Things you should know before assembly

<ol>
<li>Fastq format</li>
<li>Fasta format</li>
<li>GC- content</li>
<li>Status of genome</li>
</ol></li>
<li>Step to bacterial sequencing analysis</li>
</ul>

<h1 id="things-you-should-know-before-assembly">Things you should know before assembly</h1>

<h2 id="fastq-format">Fastq format</h2>

<p>start with <code>@</code></p>

<pre><code>@cc3e68c4-b53d-43da-be7e-b961113007e2          -&gt; sequence name
ATCCGGAATCGGTTACTGTTGGGAACCTTTGC               -&gt; sequence
+                                              -&gt; quality line break
#%(*))++.2/148447;7+001./18-7-,,30&amp;*2          -&gt; quality score
</code></pre>

<h2 id="fasta-format">Fasta format</h2>

<p>start with <code>&gt;</code></p>

<pre><code>&gt;tig00000001				           -&gt; sequence name
TGATAAAAGTATTCATATAATCTCCTATCATTTCAAAATTTAAT   -&gt; sequence 
&gt;tig00000002																	 
ATATTAGTGTGTCTATTTTATGGGGCTAGGAAAGGAGGTACATT
</code></pre>

<h2 id="gc-content">GC-content</h2>

<p>the percentage of <a href="https://en.wikipedia.org/wiki/Nitrogenous_bases">nitrogenous bases</a> on a <a href="https://en.wikipedia.org/wiki/DNA">DNA</a> or <a href="https://en.wikipedia.org/wiki/RNA">RNA</a> molecule that are either <a href="https://en.wikipedia.org/wiki/Guanine">guanine</a> or <a href="https://en.wikipedia.org/wiki/Cytosine">cytosine</a>
$$
\frac{G+C}{A+T+G+C} \times 100 \%
$$</p>

<h2 id="satuts-of-genome">Satuts of genome</h2>

<p><img src="http://ecoevo.unit.oist.jp/lab/wp-content/uploads/2013/08/GenomeAssembly.png" alt="genome status" /></p>

<h3 id="1-contig">1. contig</h3>

<p>A <strong>contig</strong>  is a set of overlapping DNA segments that together represent a <a href="https://en.wikipedia.org/wiki/Consensus_sequence">consensus region of DNA</a>.</p>

<h3 id="2-scaffold">2. scaffold</h3>

<p>To bridge the gaps between the two contigs called scaffold.</p>

<h3 id="3-complete">3. complete</h3>

<p>Didn&rsquo;t have fragment, chromesome is in one contig</p>

<p><img src="https://albertsenlab.org/wp-content/uploads/2017/11/longreadsVSshortreads.png" alt="complete genome" /></p>

<h1 id="step-to-bacterial-sequencing-analysis">Step to bacterial sequencing analysis</h1>

<p><img src="https://f1000researchdata.s3.amazonaws.com/manuscripts/14771/860b5457-c42e-40df-9829-10ecb2c4b092_figure2.gif" alt="workflow" /></p>

<h2 id="step-1-quality-control-assessing-the-quality-of-tgs">Step 1. Quality Control - Assessing the quality of TGS</h2>

<h3 id="1-1-checking-raw-read-statistics">1.1 Checking raw read statistics</h3>

<p>Tools : <a href="https://github.com/bcgsc/abyss">abyss-fac</a></p>

<pre><code>$ abyss-fac -t 1 read.fastq contig.fasta
</code></pre>

<p><img src="/img/read-abyssfac.png" alt="read-abyssfac" /></p>

<ul>
<li><p>n : Number of raw read</p></li>

<li><p>N50 : A value that present 50% of sorted read set</p></li>
</ul>

<p><img src="https://i0.wp.com/www.molecularecologist.com/wp-content/uploads/2017/03/Figure1b.jpg?w=699&amp;ssl=1" alt="Alt text" /></p>

<ul>
<li><p>Max: longest read length</p></li>

<li><p>Sum: Total read length</p></li>
</ul>

<h3 id="1-2-qc-report">1.2 QC report</h3>

<p>Tools : <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></p>

<pre><code>$ fastqc -f read.fastq -o outdir
</code></pre>

<h2 id="step-2-trimming-filtering-data-preprocessing-of-raw-data-optional">Step 2. Trimming Filtering Data - Preprocessing of raw data (optional)</h2>

<h3 id="2-1-filter-raw-read">2.1 Filter raw read</h3>

<p>Tools : <a href="https://github.com/rrwick/Filtlong">filtlong</a></p>

<p>Usually used in <strong>coverage too high</strong>, <strong>trim too short reads</strong>, and want to <strong>keep raw read in between a range</strong></p>

<h3 id="2-2-demultiplexing-trimming-adapter-barcode">2.2 Demultiplexing, Trimming adapter (barcode)</h3>

<p>Tools : <a href="https://github.com/rrwick/Porechop">porechop</a>, <a href="https://github.com/rrwick/Deepbinner">deepbinner</a></p>

<pre><code>$ porechop -i reads.fastq -o output.fastq # adapter trimming
$ porechop -i reads.fastq -b output.fastq # demultiplexing 
</code></pre>

<h2 id="step-3-sequence-assembly-long-read-genome-assembly">Step 3. Sequence Assembly - Long read genome assembly</h2>

<h3 id="3-1-de-novo-assembly">3.1 De novo assembly</h3>

<p>Tools : <a href="https://canu.readthedocs.io/en/latest/quick-start.html">Canu</a>, <a href="https://github.com/rrwick/Unicycler">Unicycler</a>, <a href="https://github.com/fenderglass/Flye">Flye</a>, <a href="https://github.com/lbcb-sci/ra">Ra</a></p>

<pre><code>$ canu -p genomename -d outdir genomeSize=4.8m -nanopore-raw read.fastq
$ unicycler -l read.fastq -o outdir
		...
</code></pre>

<p><img src="https://pic.pimg.tw/yourgene/1336720283-823145476_n.png" alt="de novo" /></p>

<h2 id="step-4-assembly-validation">Step 4. Assembly Validation</h2>

<h3 id="4-1-assembly-evaluation">4.1 Assembly evaluation</h3>

<p>Tools: <a href="https://github.com/bcgsc/abyss">abyss-fac</a>, <a href="https://github.com/sanger-pathogens/assembly-stats">assembly-stats</a></p>

<ol>
<li>Total assembly size</li>
<li>Total number of sequence</li>
<li>Longest contig</li>
<li>Average contig size</li>
<li>N50</li>
</ol>

<h3 id="4-2-assembly-status-graph">4.2 Assembly status graph</h3>

<p>Tools : <a href="https://rrwick.github.io/Bandage/">Bandage</a></p>

<p>Is it circular or linear ?</p>

<h3 id="4-3-sequence-similarity-search">4.3 Sequence Similarity Search</h3>

<p>Tools : <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a>, <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">MiGA</a></p>

<h3 id="4-4-quality-of-genomes">4.4 Quality of genomes</h3>

<p>Tools : <a href="https://ecogenomics.github.io/CheckM/">Checkm</a>, <a href="https://busco.ezlab.org/">busco</a></p>

            </div>
        </article>

        <hr />

        <div class="post-info">
                <p>
                    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-tag meta-icon"><path d="M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z"></path><line x1="7" y1="7" x2="7" y2="7"></line></svg><span class="tag"><a href="https://spaert.github.io/tags/genome-assembly">genome assembly</a></span><span class="tag"><a href="https://spaert.github.io/tags/third-generation-sequence">third generation sequence</a></span>
                </p>

            <p><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file-text"><path d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"></path><polyline points="14 2 14 8 20 8"></polyline><line x1="16" y1="13" x2="8" y2="13"></line><line x1="16" y1="17" x2="8" y2="17"></line><polyline points="10 9 9 9 8 9"></polyline></svg>348 Words</p>

            <p><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-calendar"><rect x="3" y="4" width="18" height="18" rx="2" ry="2"></rect><line x1="16" y1="2" x2="16" y2="6"></line><line x1="8" y1="2" x2="8" y2="6"></line><line x1="3" y1="10" x2="21" y2="10"></line></svg>2019-06-09 21:51 &#43;0800</p>
        </div>

        
            <div class="pagination">
                <div class="pagination__title">
                    <span class="pagination__title-h">Read other posts</span>
                    <hr />
                </div>

                <div class="pagination__buttons">
                    
                        <span class="button previous">
                            <a href="https://spaert.github.io/posts/sop-of-bacteria-sequencing-analysis/">
                                <span class="button__icon">←</span>
                                <span class="button__text">SOP of bacteria sequencing analysis</span>
                            </a>
                        </span>
                    

                    
                </div>
            </div>
        

        
    </main>

            </div>

            
                <script type="text/javascript"
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<footer class="footer">
    <div class="footer__inner">
        <div class="footer__content">
            <span>&copy; 2019</span>
            
            <span></span>
            <span> <a href="https://spaert.github.io/posts/index.xml" target="_blank" title="rss"><svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" viewBox="0 0 20 20" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-rss"><path d="M4 11a9 9 0 0 1 9 9"></path><path d="M4 4a16 16 0 0 1 16 16"></path><circle cx="5" cy="19" r="1"></circle></svg></a></span>
        </div>
    </div>
    <div class="footer__inner">
        <div class="footer__content">

        </div>
    </div>
    <div class="footer__inner">
        
    </div>
</footer>

            
        </div>

        




<script type="text/javascript" src="/bundle.min.9e52e68b082cf2a30a7fead88260edb8818fbd7f7831e39674917d4539ec75df41ba88eaddfbd916594ab4fb2a31913b46cf2d6094cf80381edb8c632512a8ca.js" integrity="sha512-nlLmiwgs8qMKf&#43;rYgmDtuIGPvX94MeOWdJF9RTnsdd9Buojq3fvZFllKtPsqMZE7Rs8tYJTPgDge24xjJRKoyg=="></script>



    </body>
</html>
